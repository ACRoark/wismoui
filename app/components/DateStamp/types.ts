export type DateStampFormat = 'auto' | 'long' | 'short';
