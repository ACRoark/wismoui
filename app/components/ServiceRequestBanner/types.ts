export interface IBannerNotification {
  hasError: boolean;
  message: IMessage;
}

export interface IMessage {
  id: string;
  values: string[] | null;
}
