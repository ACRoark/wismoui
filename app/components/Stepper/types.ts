import { ReactElement } from 'react';
import { Description } from 'types';

export interface IStepDescription {
  description: Description;
  key: string;
  status: StepperStatus;
  title: ReactElement | string;
}

export enum StepperStatus {
  Active = 'active',
  Completed = 'completed',
  Error = 'error',
  Pending = 'pending',
}
