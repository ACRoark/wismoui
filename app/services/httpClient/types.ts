export interface IHttpError extends Error {
  code: number;
  reason: string;
}

export interface IHttpResponse {
  // tslint:disable-next-line: no-any
  data: any;
  status: number;
  statusText: string;
}

export interface IHttpClient {
  cancel: () => void;
  // tslint:disable-next-line: no-any
  get(url: string, config?: any): Promise<IHttpResponse>;
  // tslint:disable-next-line: no-any
  post(url: string, data: any, config?: any): Promise<IHttpResponse>;
  // tslint:disable-next-line: no-any
  wasCancelled: (error: any) => boolean;
}
