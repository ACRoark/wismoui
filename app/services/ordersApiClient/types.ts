import { IHttpError } from 'services/httpClient';
import { IOrder } from 'types';
import IOrderSearchResults from 'types/IOrderSearchResults';
import IOrderSearchResultsViewModel from 'types/IOrderSearchResultsViewModel';
import IOrderViewModel from 'types/IOrderViewModel';

export interface IFindOrderResponse {
  error?: IHttpError;
  order: IOrder;
  viewModel?: IOrderViewModel;
}

export interface IGetOrderResponse {
  error?: IHttpError;
  order: IOrder;
  viewModel?: IOrderViewModel;
}

export interface ISearchOrdersResponse {
  error?: IHttpError;
  results: IOrderSearchResults;
  viewModel?: IOrderSearchResultsViewModel;
}
