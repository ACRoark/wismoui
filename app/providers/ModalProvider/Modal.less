@import (reference) '../../styles/themes/index.less';

@mobile: ~'(max-width: 450px)';

@modal-prefix-cls: ~'@{dte-prefix}-modal';

.@{modal-prefix-cls} {
  background: transparent !important;
  box-shadow: none !important;
  margin: 77px auto 0 !important;
  max-width: none !important;
  padding: 0px !important;
  width: 67%;

  @media @mobile {
    width: 87%;
  }

  &-body {
    background-color: @dte-white;
    border: solid 1px @dte-gray-300;
    display: flex;
    flex-direction: column;
    height: 100%;
    justify-content: space-around;
    margin-bottom: 40px;
    padding: 40px 30px;
    text-align: center;

    @media @mobile {
      padding: 40px 24px;
    }
  }

  &-close-button {
    border-color: transparent;
    color: @dte-blue-light;
    height: 20px;
    right: 0px !important;
    top: 11px !important;
    width: 20px;

    &:focus {
      outline: none;
    }
  }

  &-content {
    color: @dte-gray-600;
    font-size: @dte-wismo-font-size-0;
    text-align: left;
  }

  &-footer {
    display: flex;
    justify-content: center;
    margin-top: 10px;

    @media @mobile {
      align-items: center;
      display: flex;
      flex-direction: column;
      justify-content: center;
    }
  }

  &-overlay {
    // Known issue with React Responsive Modal that requires the overlay background to use rgba value to not apply opacity to child elements.
    background: rgba(245, 245, 245, 0.96) !important;
    padding: 0px !important;
  }

  &-title {
    color: @dte-gray-700;
    font-size: @dte-wismo-font-size-5;
    line-height: normal;
    margin: 0 35px 24px 0;
  }
}
