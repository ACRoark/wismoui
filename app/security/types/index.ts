import IAuthenticationState from 'types/IAuthenticationState';

import IAuthenticationResult from './IAuthenticationResult';

export { IAuthenticationResult };

export default IAuthenticationState;
