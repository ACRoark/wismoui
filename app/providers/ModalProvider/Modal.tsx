import classnames from 'classnames';
import React from 'react';
import { Modal as InternalModal } from 'react-responsive-modal';

import ActiveCloseIcon from 'images/ActiveCloseIcon';
import IModalConfig from 'types/IModalConfig';

import './Modal.less';

interface IModalProps {
  config: IModalConfig;
  onClose?: () => void;
}

const CloseIcon = <ActiveCloseIcon title="close" />;

const Modal: React.FC<IModalProps> = (props: IModalProps): React.ReactElement => {
  const { config, onClose } = props;
  const { className, content, footer, title } = config;

  const handleClose = (): void => {
    if (onClose) {
      onClose();
    }
  };

  return (
    <InternalModal
      classNames={{
        closeButton: 'dte-wismo-modal-close-button',
        modal: classnames(className, 'dte-wismo-modal'),
        overlay: 'dte-wismo-modal-overlay',
      }}
      closeIcon={CloseIcon}
      onClose={handleClose}
      open={!!content}
    >
      <div className="dte-wismo-modal-title">{title}</div>
      <div className="dte-wismo-modal-body">
        <div className="dte-wismo-modal-content">{content}</div>
        {footer && <div className="dte-wismo-modal-footer">{footer}</div>}
      </div>
    </InternalModal>
  );
};

export default Modal;
