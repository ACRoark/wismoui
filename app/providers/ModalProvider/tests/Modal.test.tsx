/**
 *
 * Tests for Modal
 *
 */
import { render } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import { TestRendererJsonType } from 'testing/types';

import Modal from '../Modal';

// tslint:disable-next-line: typedef
jest.mock('react-dom', () => ({
  // tslint:disable-next-line: typedef
  createPortal: node => node,
}));

const Content = (): React.ReactElement => (<div>Hello, WISMO!</div>);
const Footer = (): React.ReactElement => (<button type="button">Close</button>);
const Title = (): React.ReactElement => (<div>My Modal</div>);

describe('<Modal />', (): void => {
  it('should not log errors in console when rendered', (): void => {
    const spy = jest.spyOn(global.console, 'error');

    render(<Modal
      config={{
        content: Content,
        footer: Footer,
        title: Title,
      }} />);

    expect(spy).not.toHaveBeenCalled();
  });

  it('should render correctly when content and footer are provided', (): void => {
    const tree: TestRendererJsonType = renderer.create(
      <Modal
        config={{
          content: Content,
          footer: Footer,
        }} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when content and title are provided', (): void => {
    const tree: TestRendererJsonType = renderer.create(
      <Modal
        config={{
          content: Content,
          title: Title,
        }} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when content, footer and title are provided', (): void => {
    const tree: TestRendererJsonType = renderer.create(
      <Modal
        config={{
          content: Content,
          footer: Footer,
          title: Title,
        }} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when only content is provided', (): void => {
    const tree: TestRendererJsonType = renderer.create(
      <Modal
        config={{
          content: Content,
        }} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
