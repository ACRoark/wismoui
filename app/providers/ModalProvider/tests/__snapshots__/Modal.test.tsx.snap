// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render correctly when content and footer are provided 1`] = `
<div
  className="react-responsive-modal-overlay dte-wismo-modal-overlay"
  data-testid="overlay"
  onAnimationEnd={[Function]}
  onClick={[Function]}
  style={
    Object {
      "animation": "react-responsive-modal-fadeIn 500ms",
    }
  }
>
  <div
    aria-modal="true"
    className="react-responsive-modal-modal dte-wismo-modal"
    data-testid="modal"
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    role="dialog"
  >
    <div
      className="dte-wismo-modal-title"
    />
    <div
      className="dte-wismo-modal-body"
    >
      <div
        className="dte-wismo-modal-content"
      />
      <div
        className="dte-wismo-modal-footer"
      />
    </div>
    <button
      className="react-responsive-modal-closeButton dte-wismo-modal-close-button"
      data-testid="close-button"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          close
        </title>
        <path
          d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
          stroke="currentColor"
          strokeLinecap="round"
          strokeWidth="2"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`<Modal /> should render correctly when content and title are provided 1`] = `
<div
  className="react-responsive-modal-overlay dte-wismo-modal-overlay"
  data-testid="overlay"
  onAnimationEnd={[Function]}
  onClick={[Function]}
  style={
    Object {
      "animation": "react-responsive-modal-fadeIn 500ms",
    }
  }
>
  <div
    aria-modal="true"
    className="react-responsive-modal-modal dte-wismo-modal"
    data-testid="modal"
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    role="dialog"
  >
    <div
      className="dte-wismo-modal-title"
    />
    <div
      className="dte-wismo-modal-body"
    >
      <div
        className="dte-wismo-modal-content"
      />
    </div>
    <button
      className="react-responsive-modal-closeButton dte-wismo-modal-close-button"
      data-testid="close-button"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          close
        </title>
        <path
          d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
          stroke="currentColor"
          strokeLinecap="round"
          strokeWidth="2"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`<Modal /> should render correctly when content, footer and title are provided 1`] = `
<div
  className="react-responsive-modal-overlay dte-wismo-modal-overlay"
  data-testid="overlay"
  onAnimationEnd={[Function]}
  onClick={[Function]}
  style={
    Object {
      "animation": "react-responsive-modal-fadeIn 500ms",
    }
  }
>
  <div
    aria-modal="true"
    className="react-responsive-modal-modal dte-wismo-modal"
    data-testid="modal"
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    role="dialog"
  >
    <div
      className="dte-wismo-modal-title"
    />
    <div
      className="dte-wismo-modal-body"
    >
      <div
        className="dte-wismo-modal-content"
      />
      <div
        className="dte-wismo-modal-footer"
      />
    </div>
    <button
      className="react-responsive-modal-closeButton dte-wismo-modal-close-button"
      data-testid="close-button"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          close
        </title>
        <path
          d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
          stroke="currentColor"
          strokeLinecap="round"
          strokeWidth="2"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`<Modal /> should render correctly when only content is provided 1`] = `
<div
  className="react-responsive-modal-overlay dte-wismo-modal-overlay"
  data-testid="overlay"
  onAnimationEnd={[Function]}
  onClick={[Function]}
  style={
    Object {
      "animation": "react-responsive-modal-fadeIn 500ms",
    }
  }
>
  <div
    aria-modal="true"
    className="react-responsive-modal-modal dte-wismo-modal"
    data-testid="modal"
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    role="dialog"
  >
    <div
      className="dte-wismo-modal-title"
    />
    <div
      className="dte-wismo-modal-body"
    >
      <div
        className="dte-wismo-modal-content"
      />
    </div>
    <button
      className="react-responsive-modal-closeButton dte-wismo-modal-close-button"
      data-testid="close-button"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          close
        </title>
        <path
          d="M1 1L10.0178 10.0166M10.0178 10.0166L18.9668 18.9644M10.0178 10.0166L19 1.03558M10.0178 10.0166L1.03318 19"
          stroke="currentColor"
          strokeLinecap="round"
          strokeWidth="2"
        />
      </svg>
    </button>
  </div>
</div>
`;
